ui/src/shared/components/ConfirmButton.tsx

```diff
@@ -0,0 +1,119 @@
+import React, {PureComponent} from 'react'
+import {ClickOutside} from 'src/shared/components/ClickOutside'
+
+interface Props {
```

Comment:
all of these being conditional seems like some further refactoring would be useful

Reply:
the ?: means its an optional prop. if it has just the :, then it's required.